services/web/server/src/simcore_service_webserver/projects/_groups_api.py

```diff
@@ -139,13 +139,15 @@ async def delete_project_group(
     group_id: GroupID,
     product_name: ProductName,
 ) -> None:
-    await check_user_project_permission(
-        app,
-        project_id=project_id,
-        user_id=user_id,
-        product_name=product_name,
-        permission="delete",
-    )
+    user = await users_api.get_user(app, user_id=user_id)
+    if user["primary_gid"] != group_id:
+        await check_user_project_permission(
+            app,
```

Comment:
Q: Not sure I fully understand why `check_user_project_permission` does not include the case in which the target `group_id` is a `primary_id`

Reply:
In that case, it is the user's group, therefore he can delete himself from the project or workspace permissions even though he doesn't have write or delete permissions.